testing/integration/endpoint_security_test.go

```diff
@@ -394,17 +395,37 @@ func testInstallWithEndpointSecurityAndRemoveEndpointIntegration(t *testing.T, i
 	// Verify that the Endpoint directory was correctly removed.
 	// Regression test for https://github.com/elastic/elastic-agent/issues/3077
 	agentInstallPath := fixture.WorkDir()
-	files, err := os.ReadDir(filepath.Clean(filepath.Join(agentInstallPath, "..")))
-	require.NoError(t, err)
-
-	t.Logf("Checking directories at install path %s", agentInstallPath)
-	for _, f := range files {
-		if !f.IsDir() {
-			continue
+	agentInstallParentDir := filepath.Join(agentInstallPath, "..")
+	assert.Eventually(t, func() bool {
+		t.Logf("Checking directories at install path %s", agentInstallParentDir)
+		files, listInstallPathErr := os.ReadDir(filepath.Clean(agentInstallParentDir))
+		if err != nil {
+			t.Logf("Check failed: error reading agent install path: %s", listInstallPathErr)
+			return false
 		}
+		for _, f := range files {
+			if !f.IsDir() {
+				continue
+			}
 
-		t.Log("Found directory", f.Name())
-		require.False(t, strings.Contains(f.Name(), "Endpoint"), "Endpoint directory was not removed")
+			t.Log("Found directory", f.Name())
+			if strings.Contains(f.Name(), "Endpoint") {
+				t.Logf("Check failed: found existing endpoint directory %q", f.Name())
```

Comment:
This is what allowed us to detect what was left in the endpoint directory at the end of the test. Without it, we wouldn't know if we were left with an empty cache directory or the full endpoint install. It can be removed of course...